dapr/clients/grpc/client.py

```diff
@@ -215,13 +208,17 @@ def invoke_method(
             method (str): the method name which is called
             data (bytes or :obj:`google.protobuf.message.Message`): bytes or Message for data
                 which will send to id
-            metadata (tuple, optional): custom metadata
+            metadata (tuple, optional, DEPRECATED): custom metadata
```

Comment:
So, for service invocation specifically, I think we  need to leave the metadata parameter. Service invocation acts as a pass through for gRPC requests to an arbitrary gRPC/HTTP endpoint running on another service. That arbitrary endpoint might care about what the headers/metadata are. So I think this is the one case where we need to preserve the ability to add arbitrary gRPC metadata. 

In other APIs, we're talking directly to the dapr endpoint, which does not use gRPC metadata, and instead uses a metadata field in the request body itself. Does this distinction make sense?

In the broader sense, we may want to be deprecating the concept of calling an invoke_method endpoint entirely in favor of gRPC/HTTP proxying which we now support. This is however, a much bigger discussion that is probably out of scope of the present effort.

Reply:
This is a good concern.

I believe and lean to still deprecating this argument full stop, including for service invoke.  We still believe the use case is for informing the host of auth tokens.

I also agree we can't know what developers have already used gRPC metadata for in their own code.  Just for this scenario, and generally, I strongly recommend Dapr users adopt the gRPC and HTTP proxy approach to Service Invoke and do not use the SDK.  There will be much higher fidelity results.

For these reasons I recommend we move forward with the deprecation and monitor the situation after release and after the build warnings kick in.

Reply:
I accept this reasoning.